### PR TITLE
GH-209: Occationally, looking up react definitions fails in JSX

### PR DESCRIPTION
**Pull request by qtran-n4**
## Task
Link: https://github.com/eclipse/n4js/issues/209

### CODE REVIEW

**Comment by qtran-n4** [comment]
@mor-n4 : Please review thanks!

**Comment by qtran-n4** [comment]
I couldn't use 

```
sourceContainer.findArtifact(QualifiedName.create(REACT_MODULE, Optional.of(N4JSGlobals.N4JSD_FILE_EXTENSION))
```
as we discussed because the qualified name varies depending on where `react.n4jsd` is located.

File: plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend [hunk 1]
```diff
@@ -76,21 +70,33 @@ class ReactHelper {
...
-			val eod = descs.findFirst[ desc | desc.isReactClassifierDescription(reactClassifierFQNSuffix) ]
-			return eod?.getEObjectURI();
+			val project = n4jsCore.findProject(context.eResource.URI).get;
+
+			var URI reactURI = null
+			// Lookup react.n4jsd in source folders
+			for (sourceContainer : project.sourceContainers) {
+				if (reactURI === null) {
+					// srcContainer is an iterable URIs
+					val ret = sourceContainer.findFirst[trimFragment.lastSegment == REACT_DEFINITION_FILE]
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend]: you can directly assign to `reactURI` (no need for variable `ret`)

File: plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend [hunk 2]
```diff
@@ -76,21 +70,33 @@ class ReactHelper {
...
+					if (ret !== null) {
+						reactURI = ret
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend]: using `break;` would be nicer (or isn't this supported in Xtext?)

File: plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/typesystem/N4JSXUnsupportedExpressionTypeHelper.xtend [hunk 3]
```diff
@@ -40,15 +40,13 @@ class N4JSXUnsupportedExpressionTypeHelper extends DefaultUnsupportedExpressionT
 	override public typeExpression(Expression expression, RuleEnvironment G) {
 		if (expression instanceof JSXElement) {
 			val classifierReactElement = reactLookupHelper.lookUpReactElement(expression);
-
-			if (classifierReactElement === null) {
-				throw new IllegalStateException(ReactHelper.REACT_MODULE + "." + ReactHelper.REACT_ELEMENT +  " not found");
+			if (classifierReactElement !== null) {
+				val typeRef = TypeUtils.createTypeRef(classifierReactElement)
+				return typeRef
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/typesystem/N4JSXUnsupportedExpressionTypeHelper.xtend]: why this change?

File: plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend (hunk 1 above)
Comment (qtran-n4) [reply plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend]: done

File: plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend (hunk 2 above)
Comment (qtran-n4) [reply plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend]: Xtend does not support `break;`

**Comment by qtran-n4** [comment]
Thanks for your review! Builds are green so merging.
